SCClassLibrary/JITLib/ProxySpace/extStoreOn.sc

```diff
@@ -344,14 +344,28 @@
 
 	asCode { | namestring = "", dropOut = false |
 		^String.streamContents({ |stream|
-			var map;
+			var map, rate;
 			if(dropOut) {
 				map = this.copy;
 				map.removeAt(\out);
 				map.removeAt(\i_out);
 			} { map = this };
 
 			if(map.notEmpty) {
+				// 'map' might refer to other NodeProxies
+				// Before referring to them in an arg list,
+				// we should be sure they are initialized
+				// to the right rate and number of channels.
+				// It is OK if the 'source' definition comes later in the doc string.
+				map.keysValuesDo { |key, value|
+					if(value.isKindOf(BusPlug)) {
+						rate = value.rate;
+						if(rate != \audio) { rate = \control };
+						stream <<< value  // storeOn gets the ~key
+						<< "." << UGen.methodSelectorForRate(rate)
```

Comment:
so that means the resulting string in these cases will look like `p1.set(p2.ar(2), p3.kr(1))`?